### PR TITLE
Adding support for RatingInput and ReadOnly Rating element 

### PR DESCRIPTION
**Pull request by wyAbhishek**
# Support for New Rating elements

This PR adds support for two new elements: "Input.Rating" and "Rating".  First one is for taking rating value as input from users and second one shows rating value in read only mode. 

# Spec for Input.Rating:

| Property name       | Required | Type                                             | Allowed values                                  | Description                                                                     |
| ------------------- | -------- | ------------------------------------------------ | ----------------------------------------------- | ------------------------------------------------------------------------------- |
| type                | Yes      | String, must be Input.Rating                     |                                                 |                                                                                 |
| id                  | Yes      | String                                           |                                                 |                                                                                 |
| requires            |          | Object                                           |                                                 |                                                                                 |
| isVisible           |          | Boolean                                          |                                                 |                                                                                 |
| separator           |          | Boolean                                          |                                                 |                                                                                 |
| height              |          | String                                           | Auto, Stretch                                   |                                                                                 |
| horizontalAlignment |          | String                                           | Left, Center, Right                             |                                                                                 |
| spacing             |          | String                                           | None, Small, Default, Medium, Large, ExtraLarge |                                                                                 |
| targetWidth         |          | String                                           | VeryNarrow, Narrow, Standard, Wide              |                                                                                 |
| label               |          | String                                           |                                                 |                                                                                 |
| isRequired          |          | Boolean                                          |                                                 |                                                                                 |
| errorMessage        |          | String                                           |                                                 |                                                                                 |
| valueChangedAction  |          | Action object, can only be an Action.ResetInputs |                                                 |                                                                                 |
| value               |          | Number                                           |                                                 | The number of stars selected by default. Cannot exceed max if max is specified. |
|                     |          |                                                  |                                                 |                                                                                 |
|                     |          | SMBA: just ensure it’s a number                  |                                                 |                                                                                 |
| max                 |          | Number                                           |                                                 | The total number of stars. Maximum and default is 5.                            |
|                     |          |                                                  |                                                 |                                                                                 |
|                     |          | SMBA: just ensure it’s a number                  |                                                 |                                                                                 |
| allowHalfSteps      |          | Boolean                                          |                                                 | Determines if the user can select half stars                                    |
| size                |          | String                                           | Medium or Large                                 | The size of the stars                                                           |
| color               |          | String                                           | Neutral or Marigold                             | The color of the stars                                                          |

# Spec for Read Only Rating:
| Property name       | Required | Type                   | Allowed values                                  | Description                                                                             |
| ------------------- | -------- | ---------------------- | ----------------------------------------------- | --------------------------------------------------------------------------------------- |
| type                | Yes      | String, must be Rating |                                                 |                                                                                         |
| id                  |          | String                 |                                                 |                                                                                         |
| requires            |          | Object                 |                                                 |                                                                                         |
| isVisible           |          | Boolean                |                                                 |                                                                                         |
| separator           |          | Boolean                |                                                 |                                                                                         |
| height              |          | String                 | Auto, Stretch                                   |                                                                                         |
| horizontalAlignment |          | String                 | Left, Center, Right                             |                                                                                         |
| spacing             |          | String                 | None, Small, Default, Medium, Large, ExtraLarge |                                                                                         |
| targetWidth         |          | String                 | VeryNarrow, Narrow, Standard, Wide              |                                                                                         |
| value               |          | Number                 |                                                 | The number of stars selected. Cannot exceed max if max is specified.                    |
| count               |          | Number                 |                                                 | The number of ratings that make up the average (see the gold star in the example above) |
| max                 |          | Number                 |                                                 | The total number of stars. Maximum and default is 5.                                    |
| size                |          | String                 | Medium or Large                                 | The size of the stars                                                                   |
| color               |          | String                 | Neutral or Marigold                             | The color of the stars                                                                  |
| style               |          | String                 | Default or Compact                              | In compact mode, only one star is displayed.                                            |

For all Pull Requests, please describe how the issue was fixed or how the feature was implemented from a summary level. This information will be used to help provide context to the reviewers of the pull request and should be additive to the issue being closed.

# Sample Card
Simulator Screenshot - iPhone 15 Plus - 2024-05-22 at 10 01 55

### CODE REVIEW

File: source/ios/AdaptiveCards/AdaptiveCards/AdaptiveCards/ACRRatingInputRenderer.mm [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+           rootView:(ACRView *)rootView
+             inputs:(NSMutableArray *)inputs
+    baseCardElement:(ACOBaseCardElement *)acoElem
+         hostConfig:(ACOHostConfig *)acoConfig;
+{
+    std::shared_ptr<HostConfig> config = [acoConfig getHostConfig];
+    std::shared_ptr<BaseCardElement> elem = [acoElem element];
+    std::shared_ptr<RatingInput> ratingInput = std::dynamic_pointer_cast<RatingInput>(elem);
+    
+    NSInteger maxValue = ratingInput->GetMax() > 5 ? 5: ratingInput->GetMax();
```
Comment (GabrielMedAlv) [inline source/ios/AdaptiveCards/AdaptiveCards/AdaptiveCards/ACRRatingInputRenderer.mm:46]: Can you put the "5" as a constant instead of hardcoded?

**Review by GabrielMedAlv — COMMENTED** [review]
Please include some unt tests for this feature.

File: source/ios/AdaptiveCards/AdaptiveCards/AdaptiveCards/ACRRatingInputRenderer.mm (hunk 1 above)
Comment (wyAbhishek) [reply source/ios/AdaptiveCards/AdaptiveCards/AdaptiveCards/ACRRatingInputRenderer.mm:46]: I couldn't find any constants file in codebase. Literal values like these are used everywhere in code. Do you want me to create a new constants file for this?

**Comment by wyAbhishek** [comment]
> Please include some unt tests for this feature.

Added tests for validation of properties for RatingLabel and RatingInput

**Review by GabrielMedAlv — APPROVED** [review]

File: source/ios/AdaptiveCards/AdaptiveCards/AdaptiveCards/ACRRatingInputRenderer.mm (hunk 1 above)
Comment (GabrielMedAlv) [reply source/ios/AdaptiveCards/AdaptiveCards/AdaptiveCards/ACRRatingInputRenderer.mm:46]: You are right. Then is ok as it is.

**Comment by prpatwa** [comment]
Suggestion - Can we have separate renderer and view for read only and normal rating element to prevent any regressive change in readonly rating element. Considering the cases where other element might trigger value change in read only rating element.

**Review by prpatwa — APPROVED** [review]

**Pull request merged**